storage/core/src/server/ServerStorage.ts

```diff
@@ -52,6 +52,8 @@ export abstract class ServerStorage
 
   public abstract createBaseDirectory(directory: BaseDirectory): Promise<void>;
 
+  public abstract listDirectories(): Promise<BaseDirectory[]>;
+
   public abstract list(directory: BaseDirectory): Promise<ObjectReference[]>;
```

Comment:
I suggest we add a method `listObjects` and mark `list` as deprecated because now it looks confusing when there are `list` and `listDirectories` methods. @Animagne thoughts?

Reply:
But in that case, we would want to have list directories be able to take a base directory as an argument, right?

Reply:
no, the proposed interface is as follows:
```
public abstract listDirectories(): Promise<BaseDirectory[]>;
public abstract listObjects(directory: BaseDirectory): Promise<ObjectReference[]>;

@deprecated
public abstract list(directory: BaseDirectory): Promise<ObjectReference[]>;
```

The only change here is essentially to rename `list` to be more specific to match `listDirectories`. `listDirectories` would not have to change its interface. This is purely for readability, not functionality.

Reply:
No objection, I'm making the change

Reply:
I see you have removed `list` completely and added `listObjects`, in other words, you have renamed `list` to `listObjects`. We cannot do that since this is public API and is a breaking change. Instead we should keep the old method, mark it as deprecated, and add a new method with a new name that would behave identically to the old method.

Reply:
I see you have removed `list` completely. We cannot do that as this is public API and removing methods is a breaking change. Instead we should keep the old method, mark it as deprecated and add a new method with the new name that behaves identically to the old one. This way new users will use the better named method and the code won't break for the old users that are already using the `list` method.